src/coreclr/jit/flowgraph.cpp

```diff
@@ -6787,6 +6894,398 @@ void FlowGraphDominanceFrontiers::Dump()
+            m_preorderNum[postDominated->bbReversePostorderNum]) &&
+           (m_postorderNum[postDominator->bbReversePostorderNum] >=
+            m_postorderNum[postDominated->bbReversePostorderNum]);
+}
+
+#ifdef DEBUG
+//------------------------------------------------------------------------
+// FlowGraphPostDominatorTree::Dump: Dump a textual representation of the postdominator
+// tree.
+//
+void FlowGraphPostDominatorTree::Dump()
+{
+    Compiler* comp = m_reverseDfsTree->GetCompiler();
+
+    for (BasicBlock* block : comp->Blocks())
+    {
+        if (!m_reverseDfsTree->Contains(block) || (m_domTree[block->bbReversePostorderNum].firstChild == nullptr))
+            continue;
+
+        printf(FMT_BB " : ", block->bbNum);
+        for (BasicBlock* child = m_domTree[block->bbReversePostorderNum].firstChild; child != nullptr;
+             child             = m_domTree[child->bbReversePostorderNum].nextSibling)
+        {
+            printf(FMT_BB " ", child->bbNum);
+        }
+        printf("\n");
+    }
+
+    printf("\n");
+}
+#endif
+
+//------------------------------------------------------------------------
+// FlowGraphPostDominatorTree::Build: Compute the dominator tree for the blocks in
+// the DFS tree.
+//
+// Parameters:
+//   reverseDfsTree - reverse DFS tree.
+//
+// Returns:
+//   Data structure representing postdominator tree. Immediate postdominators are
+//   marked directly into the BasicBlock structures, in the bbIPDom field, so
+//   multiple instances cannot be simultaneously used.
+//
+// Remarks:
+//   As a precondition it is required that the reverse flow graph has a unique root.
+//   This is handled by the reverse DFS tree via its pseudo block.
+//
+FlowGraphPostDominatorTree* FlowGraphPostDominatorTree::Build(const FlowGraphReverseDfsTree* reverseDfsTree)
+{
+    Compiler*         comp       = reverseDfsTree->GetCompiler();
+    BasicBlock**      postOrder  = reverseDfsTree->GetPostOrder();
+    unsigned          count      = reverseDfsTree->GetPostOrderCount();
+    BasicBlock* const pseudoExit = reverseDfsTree->PseudoExit();
+
+    pseudoExit->bbIPDom = nullptr;
+
+    // First compute immediate postdominators.
+    unsigned numIters = 0;
+    bool     changed  = true;
+    while (changed)
+    {
+        changed = false;
+
+        // In reverse post order, except for the entry block (count - 1 is entry BB).
+        for (unsigned i = count - 1; i > 0; i--)
+        {
+            unsigned    poNum = i - 1;
+            BasicBlock* block = postOrder[poNum];
+
+            // Intersect Postdom, if computed, for all successors
+            BasicBlock* bbIPDom = nullptr;
+
+            auto visitSucc = [=, &bbIPDom](BasicBlock* succ) {
+                if (!reverseDfsTree->Contains(succ))
+                {
+                    // Unreachable succ...?
+                    return;
+                }
+
+                if ((numIters <= 0) && (succ->bbReversePostorderNum <= poNum))
```

Comment:
I did not understand the meaning of check `numIters <= 0`. It is unsigned, so it will be `>= 0`. Perhaps this should be `numIters == 0`? I see similar thing in `FlowGraphDominatorTree::Build` too.

Reply:
Yes, they are equivalent (I for some reason usually write this check in this way)